### ISSUE TITLE
[Bug]: Pre-release CI fails with pandas future deprecation warning

### ISSUE THREAD

**Issue by rly** [hdmf-dev/hdmf#761]
### What happened?

The nightly pre-release CI fails because using the pre-release version of pandas results in the following pandas FutureWarning in the gallery tests:
```
FutureWarning: The 'inplace' keyword in DataFrame.set_index is deprecated and will be removed in a future version. Use `df = df.set_index(..., copy=False)` instead.
```

This occurs in:
1. /src/hdmf/common/alignedtable.py
2. docs/gallery/plot_external_resources.py

### Steps to Reproduce

```shell
Run tests
```

### Traceback

```shell
Traceback (most recent call last):
  File "/home/runner/work/hdmf/hdmf/test_gallery.py", line 73, in run_gallery_tests
    _import_from_file(script)
  File "/home/runner/work/hdmf/hdmf/test_gallery.py", line 24, in _import_from_file
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/runner/work/hdmf/hdmf/docs/gallery/plot_external_resources.py", line 506, in <module>
    table.set_index('id', inplace=True)
  File "/home/runner/work/hdmf/hdmf/.tox/gallery-py310-prerelease/lib/python3.10/site-packages/pandas/util/_decorators.py", line 317, in wrapper
    return func(*args, **kwargs)
  File "/home/runner/work/hdmf/hdmf/.tox/gallery-py310-prerelease/lib/python3.10/site-packages/pandas/core/frame.py", line 5986, in set_index
    warnings.warn(
FutureWarning: The 'inplace' keyword in DataFrame.set_index is deprecated and will be removed in a future version. Use `df = df.set_index(..., copy=False)` instead.
```

### Operating System

Windows

### Python Executable

Python

### Python Version

3.10

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/hdmf-dev/hdmf/blob/dev/.github/CODE_OF_CONDUCT.md)
- [X] Have you checked the [Contributing](https://github.com/hdmf-dev/hdmf/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this bug was not already [reported](https://github.com/hdmf-dev/hdmf/issues)?

**Comment by oruebel** [comment]
@rly I added fixes for this in #762 The annoying thing is that the function signature of ``pandas.DataFrame.set_index`` changes in a manner that doesn't allow us to set an index inplace (without copying the data) in the same way with pandas <1.4 and >1.5. In the galleries we can avoid this issue by just copying the table, which has the same signature across the pandas versions. In alignedtable.py, however, we should avoid copying the data so there I added a check to use the appropriate variant depending on the pandas version. However, once we require Pandas >=1.5 we should remove those checks.
